pkg/defaults/defaults.go

```diff
@@ -0,0 +1,157 @@
+package defaults
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/operator-framework/operator-marketplace/pkg/apis/operators/v1"
+	wrapper "github.com/operator-framework/operator-marketplace/pkg/client"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/api/errors"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/yaml"
+)
+
+var (
+	// Dir is the directory where the default OperatorSource definitions are
+	// placed on disk. It will be empty if defaults are not required.
+	Dir string
```

Comment:
non-blocking: instead of a global `dir` for the package, `defaults` could have a `dir` member. In the future if we needed multiple types of defaults to apply we can just make a new default tracker.

Reply:
Should be easy enough to do. I anyhow need to rebase now that #228 has merged. Let me see if I can quickly do it or I will leave it as it is.

Reply:
Shoot. I forgot to put a hold on the PR and it merged. Commit history now looks a bit weird with two commits fixing the `isEqual()`. Sorry about that. I have already made the change you requested and will add it to the next PR.